File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/ClientUnitFactory.java
package io.appmetrica.analytics.impl.component.clients;

import android.content.Context;
import androidx.annotation.NonNull;
import io.appmetrica.analytics.impl.component.CommonArguments;

public interface ClientUnitFactory<C extends ClientUnit> {

    @NonNull
    C createClientUnit(@NonNull Context context,
                       @NonNull ComponentsRepository repository,
                       @NonNull ClientDescription clientDescription,
                       @NonNull CommonArguments sdkConfig);
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/component/clients/ClientUnitFactory.kt
package io.appmetrica.analytics.impl.component.clients

import android.content.Context
import io.appmetrica.analytics.impl.component.CommonArguments

internal interface ClientUnitFactory<C : ClientUnit> {
    fun createClientUnit(
        context: Context,
        repository: ComponentsRepository,
        clientDescription: ClientDescription,
        sdkConfig: CommonArguments
    ): C
}
